src/Core/Config/ValueObject/Cache.php

```diff
@@ -159,40 +160,49 @@ public function getDataBySource(int $source): array
 	/**
 	 * Sets a value in the config cache. Accepts raw output from the config table
 	 *
-	 * @param string $cat    Config category
-	 * @param string $key    Config key
-	 * @param mixed  $value  Value to set
-	 * @param int    $source The source of the current config key
+	 * @param string  $cat    Config category
+	 * @param ?string $key    Config key
+	 * @param ?mixed  $value  Value to set
+	 * @param int     $source The source of the current config key
 	 *
 	 * @return bool True, if the value is set
 	 */
-	public function set(string $cat, string $key, $value, int $source = self::SOURCE_DEFAULT): bool
+	public function set(string $cat, string $key = null, $value = null, int $source = self::SOURCE_DEFAULT): bool
 	{
-		if (!isset($this->config[$cat])) {
+		if (!isset($this->config[$cat]) && $key !== null) {
 			$this->config[$cat] = [];
 			$this->source[$cat] = [];
 		}
 
-		if (isset($this->source[$cat][$key]) &&
-			$source < $this->source[$cat][$key]) {
+		if ((isset($this->source[$cat][$key]) && $source < $this->source[$cat][$key]) ||
+			(is_int($this->source[$cat] ?? null) && $source < $this->source[$cat])) {
 			return false;
 		}
 
 		if ($this->hidePasswordOutput &&
 			$key == 'password' &&
 			is_string($value)) {
-			$this->config[$cat][$key] = new HiddenString((string)$value);
+			$this->setCatKeyValueSource($cat, $key, new HiddenString((string)$value), $source);
```

Comment:
`$value` already is a string per condition.
```suggestion
			$this->setCatKeyValueSource($cat, $key, new HiddenString($value), $source);
```